### PR TITLE
* Unit Test for CartController

### PR DESCRIPTION
**Pull request by dangenendt**
* removed duplicate unnecessary code from "addToCart"-method. Tax calculation was done twice
* added unit test for CartController

### CODE REVIEW

**Comment by dangenendt** [comment]
Hmmm... where are those binary files are coming from? Didnt touched them :confused:

File: app/Http/Controllers/CartController.php [hunk 1]
```diff
@@ -54,29 +54,8 @@ public function addToCart(Request $request)
 
         Cart::add($slug, $qty, $attribute);
 
-
         $this->_setTaxAmount($slug, $qty);
```
Comment (indpurvesh) [inline app/Http/Controllers/CartController.php]: First of all sorry seems very complicated code.  

    $this->setTaxAmount($slug, $qty, $attribute);

And you want to replace `setTaxAmount`method as below:

     /**
     * Set the Tax Amount for the Product
     * 
     * @param string $slug
     * @param int $qty
     * @param array $attributes
     * @return self $this
     */
    private function _setTaxAmount($slug, $qty = 1, $attributes = [])
    {
        $productModel = $this->repository->findBySlug($slug);
        $isTaxEnabled = $this->configurationRepository->getValueByKey('tax_enabled');

        if ($isTaxEnabled && $productModel->is_taxable) {
            $percentage = $this->configurationRepository->getValueByKey('tax_percentage');
            
            if(null !== $attributes) {
                foreach($attributes as $attributeId => $productId) {
                    $productModel = $this->repository->find($productId);
                }
            }
            
            $taxAmount = ($percentage * $productModel->price / 100) * $qty;
            Cart::hasTax(true);
            Cart::updateProductTax($slug, $taxAmount);
        }

        return $this;
    }


Controller should look like below

    /**
     *
     * Add To Cart Product
     *
     * @param Request $request
     * @return \Illuminate\Http\RedirectResponse
     */
    public function addToCart(Request $request)
    {
        $slug = $request->get('slug');

        $qty = abs($request->get('qty', 1));
      
        $attribute = $request->get('attribute', null);

        if (!Cart::canAddToCart($slug, $qty, $attribute)) {
            return redirect()->back()
                        ->with(
                            'errorNotificationText',
                            'Not Enough Qty Available. Please with less qty or Contact site Administrator!'
                        );
        }

        Cart::add($slug, $qty, $attribute);
        $this->setTaxAmount($slug, $qty);

        return redirect()->back()->with('notificationText', 'Product Added to Cart Successfully!');
    }

**Comment by indpurvesh** [comment]
@dangenendt 

I think your unit test still should run perfectly fine.

One more thing for some reason you are deleting the dummy image files in PR as well. so you want to ignore that as well.

Thanks for the nice work.

**Comment by indpurvesh** [comment]
It should not delete the files ever but it just copies the files to the `storage/app/public/uploads` folder. this is what the installation process does. 

If you want you can close this PR and do another PR with only UnitTest file? I will update the cart controller?

**Comment by dangenendt** [comment]
No no, i will do that. 
I close this PR and create a new one

**Pull request closed without merging**